link

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env php
+<?php
+
+/*
+* This file is part of the Symfony package.
+*
+* (c) Fabien Potencier <fabien@symfony.com>
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+/**
+ * Link dependencies to components to a local clone of the monolithic Symfony repository.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+
+if ($argc < 2) {
+    echo 'Link dependencies to components to a local clone of the monolithic Symfony repository.'.PHP_EOL.PHP_EOL;
+    echo 'Usage: ./link /path/to/the/project'.PHP_EOL;
+    exit(1);
+}
+
+if (!is_dir("$argv[1]/vendor/symfony")) {
+    echo "The directory \"$argv[1]\" does not exist or the dependencies are not installed, did you forget to run \"composer install\" in your project?";
+    exit(1);
+}
+
+$sfPackages = array();
+foreach (glob(__DIR__.'/src/Symfony/{Bundle,Bridge,Component}/*', GLOB_BRACE | GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+    $sfPackages[json_decode(file_get_contents("$dir/composer.json"))->name] = $dir;
+}
+
+foreach (glob("$argv[1]/vendor/symfony/*", GLOB_BRACE | GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+    if (is_link($dir)) {
+        echo "\"symfony/$package\" is already a symlink, skipping.".PHP_EOL;
+        continue;
+    }
+
+    $package = basename($dir);
+    if (isset($sfPackages["symfony/$package"])) {
+        rename($dir, "$dir.back");
```

Comment:
intended as `.bak`? https://en.wikipedia.org/wiki/Bak_file

Can/should we have an option to skip backup files, so no double classes are indexed in Phpstorm and such.

Reply:
I don't care about the backup too.
Composer install is really fast when all packages are in cache

Reply:
Same, I don't see the point of the backup. I'm always doing a `rm -rf vendor/symfony/*.back` as it also messed up PhpStorm...

Reply:
It's not for the speed, it's just in case the developper has done some changes in the vendor (common when trying to debug something). Ok for the `--no-backup` flag.

Reply:
but this fails when run twice

Reply:
But I don't see why you'd do that 😄

Reply:
Oh... maybe after adding another dependency...

Reply:
> but this fails when run twice

No because the second time it will be skipped (the symlink will already exist)

Reply:
Anyway I'm ok to remove the backup feature nobody except me find this hazardous :)

Reply:
drop :)

Reply:
done